### PR TITLE
Check for known subsystem against consumers and subscribers

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Updates https://github.com/kolide/launcher/pull/1780

Addresses comment thread: https://github.com/kolide/launcher/pull/1780#discussion_r1680233506

### CODE REVIEW

File: ee/control/control.go [hunk 1]
```diff
@@ -366,20 +364,30 @@ func (cs *ControlService) fetchAndUpdate(subsystem, hash string) error {
...
 }
 
+// knownSubsystem checks our registered consumers and subscribers to see if the given
+// subsystem is one that has been registered with the control service.
+func (cs *ControlService) knownSubsystem(subsystem string) bool {
+	if _, subsystemFromConsumer := cs.consumers[subsystem]; !subsystemFromConsumer {
+		if _, subsystemFromSubscriber := cs.subscribers[subsystem]; !subsystemFromSubscriber {
+			return false
+		}
+	}
```
Comment (directionless) [inline ee/control/control.go:374]:
```suggestion
	if _, subsystemFromConsumer := cs.consumers[subsystem]; subsystemFromConsumer {
	  return true
	}
	if _, subsystemFromSubscriber := cs.subscribers[subsystem]; subsystemFromSubscriber {
			return true
	}
	
	return false
```

I think? But it seems fine either way

**Review by directionless — APPROVED** [review]

File: ee/control/control.go (hunk 1 above)
Comment (RebeccaMahany) [reply ee/control/control.go:374]: Oh yup, reads better without the additional nesting -- updating now

File: ee/control/control_test.go [hunk 2]
```diff
@@ -410,6 +410,71 @@ func TestControlServicePersistLastFetched(t *testing.T) {
 	}
 }
 
+func Test_knownSubsystem(t *testing.T) {
```
Comment (RebeccaMahany) [inline ee/control/control_test.go:413]: This is also tested within the context of `Fetch` in `TestControlServiceFetch_IgnoresUnknownSubsystems`

**Review by directionless — APPROVED** [review]
I think cleaner. Yay

**Pull request merged**